제출자 최현준입니다.

che0223 opened the pull request:
내용이 좀 어려워서 제출이 늦었습니다(_ _);; 언제나 좋은 강의 감사드립니다:)

che0223 committed:
1st commit

create

Moon-edu commented on homework/dags/my_dag.py:
```diff
@@ -1,67 +1,82 @@
+            with conn.cursor() as cur:
+                cur.executemany(f"""
+                    insert into stock(code, name, price, price_date, category) values(%s, %s, %s, %s, %s);
+                """, list_stock_data)
```
https://www.psycopg.org/docs/cursor.html#cursor.executemany
psycopg2 공식문서에 따르면, executemany는 효율적이지 못하다는 경고가 있습니다.

che0223 replied:
execute_batch(cur, "INSERT INTO test (num, data) VALUES (%s, %s)", tuples) 로 바꿔서 시도해보겠습니다:)

Moon-edu commented on homework/dags/my_dag.py:
```diff
@@ -1,67 +1,82 @@
+    csv_downloader = BashOperator(
+        task_id='download_csv',
+        bash_command=f'curl -k -o {CSV_OUTPUT_PATH} '
+                     + f'{CSV_ENDPOINT}')
```
좀 사소한거긴한데 이걸 다른 줄에 단 이유가 있나요?

che0223 replied:
sample 코드에서 BashOperator 부분을 복붙하다보니 발생한 문제같습니다 ;

Moon-edu commented:
잘하셨습니다. `executemany`빼고는 따로 피드백드릴건 없습니다.

che0223 committed:
2nd commit

1. 코드 수정
import psycopg -> import psycopg2

cur.executemany() -> psycopg2.extras.execute_values()

requirements=["psycopg==3.1.4", "xmltodict==0.13.0"]
->
requirements=["psycopg2==2.9.1", "xmltodict==0.13.0"]

che0223 commented:
https://brownbears.tistory.com/567 사이트에서 보니까 unnest 라는 방법이 1.5배 정도 더 빠른거 같긴 한데, 값 하나마다 리스트를 만들어야 하는거 같아서, 값을 뭉쳐서 만들어야하는 이 문제에서는 execute_batch가 더 효율적이라 생각되어 execute_batch를 사용했습니다.
